### PR TITLE
Fix type deduction bug

### PR DESCRIPTION
**Pull request by amadio**
Please see commit messages for more information. This probably also fixes [ROOT-9139](https://sft.its.cern.ch/jira/browse/ROOT-9139).

### CODE REVIEW

File: core/base/inc/Rtypes.h [hunk 1]
```diff
@@ -253,7 +233,7 @@ class ClassDefGenerateInitInstanceLocalInjector:
       static const char* Name() {
          static std::string gName;
          if (gName.empty())
-            SetName(TTypeNameExtraction<T>::Get(), gName);
+            SetName(GetDemangledName(typeid(T).name()), gName);
```
Comment (pcanal) [inline core/base/inc/Rtypes.h]: Rather than creating a temporary std::string, can we pass a const char* or better yet the type_info (& or *) ?

Comment (amadio) [reply core/base/inc/Rtypes.h]: I tried to maintain the same interfaces more or less. I will change to pass the `type_info` directly and then merge this into master.

**Comment by amadio** [comment]
Failed report is not related to this PR. I will merge manually soon.

**Comment by amadio** [comment]
Failed to check because it was not there (was merged into master and branch was deleted...).
Failing test (dataframe reports) is unrelated and probably now fixed.

**Pull request merged**